### ISSUE TITLE
openapi2kong should allow multiple "foreign key" plugins

### ISSUE THREAD

**Issue by mttrhll** [Kong/go-apiops#183]
In Konnect some of our "gateway services" use many instances of the rate-limiting-advanced plugin and have multiple "foreign keyed" resources. This is a helpful architecture for doing things like rate limiting tiers for a single service. 

Example would be an RLA that is "scoped" to: consumerA + serviceA. Another instance of an RLA is "scoped" to: consumerB + serviceA. There does not seem to be a way to do this using openapi2kong. It seems to try to "promote" both and the second one fails. This issue seems to happen even with 2 different plugins that have consumers.

To replicate add two RLA plugins that are configured with "consumers" to two paths in an openApi file.

Simply uncommenting the consumer on 1 of the 2 allows the command to succeed. I'm using:

deck 1.38 on a Mac. The command issued is pretty standard:  
  deck file openapi2kong -s <openApi-file.yaml> -o <output-file.yaml> --inso-compatible

FWIW - it would be great if "root" level plugins could be put into the openApi file using some "x-kong-plugins:" or similar. The final structure in deck for these types of configurations look like:

 ...
plugins: <-- at the root of the deck file
- config:
     limit:
     - 5
     ...
   consumer: consumerA
   service: serviceA
   name: rate-limiting-advanced
- config:
     limit:
     - 10
     ...
   consumer: consumerB
   service: serviceA
   name: rate-limiting-advanced

Error produced:
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/kong/go-apiops/openapi2kong.Convert.func1(0x1000100?, 0x14bf62588?)
        github.com/kong/go-apiops@v0.1.33/openapi2kong/openapi2kong.go:1126 +0x1d0
sort.insertionSort_func({0x14000123a90?, 0x140003ae080?}, 0x0, 0x3)
        sort/zsortfunc.go:12 +0xc0
sort.pdqsort_func({0x14000123a90?, 0x140003ae080?}, 0x140001975a8?, 0x0?, 0x0?)
        sort/zsortfunc.go:73 +0x260
sort.Slice({0x103d9db20?, 0x140001975a8?}, 0x14000123a90)
        sort/slice.go:29 +0xbc
github.com/kong/go-apiops/openapi2kong.Convert({0x14000174000, 0x769e, 0x769f}, {{0x0, 0x0, 0x0}, {0x0, 0x0}, {0x6b, 0xa7, ...}, ...})
        github.com/kong/go-apiops@v0.1.33/openapi2kong/openapi2kong.go:1122 +0x1a18
github.com/kong/deck/cmd.executeOpenapi2Kong(0x140001bcb00?, {0x1400010bef0?, 0x4?, 0x103a49a2a?})
        github.com/kong/deck/cmd/file_openapi2kong.go:61 +0x3f4
github.com/spf13/cobra.(*Command).execute(0x14000133808, {0x1400010bea0, 0x5, 0x5})
        github.com/spf13/cobra@v1.8.0/command.go:983 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x140001ba008)
        github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.8.0/command.go:1032
github.com/kong/deck/cmd.Execute({0x1040d8b10, 0x140003eeaf0})
        github.com/kong/deck/cmd/root.go:265 +0x64
main.main()
        github.com/kong/deck/main.go:31 +0x20

**Comment by Tieske** [comment]
what configuration are you providing to `deck file openapi2kong`?

**Comment by mttrhll** [comment]
I created a simple oas yaml file to reproduce the issue. The deck cmd works if the consumer is commented out of 1 of the x-kong-plugin-rate-limiting-advanced configurations.

I also put in the second scenario where different plugins can cause the same issue if they both have consumers.
[openapi-issue-183.yaml.txt](https://github.com/user-attachments/files/16087052/openapi-issue-183.yaml.txt)

Of note, I used the kong ui to actually create scoped plugins like this and then dumped them. They were in the root of the deck file under plugins.... just like the code seems to intend for "promoted" plugin configs.

Note: I had to name the example file .txt to attach it for some reason.
